src/couch_index/src/couch_index_plugin_couch_db.erl

```diff
@@ -0,0 +1,42 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_index_plugin_couch_db).
+
+-export([
+    before_copy_purge_info/1
+]).
+
+-include_lib("couch/include/couch_db.hrl").
+-include_lib("couch_mrview/include/couch_mrview.hrl").
+
+
+before_copy_purge_info(DbName) ->
+    {ok, DDocs} = design_docs(DbName),
+    lists:map(fun(DDoc) ->
+        JsonDDoc = couch_doc:from_json_obj(DDoc),
+        couch_mrview_index:maybe_create_local_purge_doc(DbName, JsonDDoc)
```

Comment:
Understand that we had better not to use `couch_mrview_index` in `couch_index` repository. I also tried with different approach. But didn't find more graceful way. Alternatively, I can try to make it configurable and open to any suggestions.

The difficulty here is that this was called from https://github.com/apache/couchdb/pull/1187/commits/c5eb57ba4929182d46cd23973b37d895441e71c6#diff-6b8890f55c19eb040b25463764ed0a05R64 where there is no context information about `couch_mrview_index`.  This is different calls triggered from `couch_mrview` repository and couch_mrview_index can be passed as `Mod`.

Reply:
But why does it even need to be in `couch_index`? I mean it's a plugin, it could live in any repo, even in a private one.

Reply:
Yes, the plugin could live in any repo, even in a private one. The point is here related to design patten about couch_index and couch_mrview. They are bundled for mrview index. I ever try to add logic to couch_mrview_epi, couch_mrview_plugin and couch_mrview_plugin_couch_db. However, we still need to register couch_mrview_epi. Can't find right place in couch_mrview, but https://github.com/apache/couchdb/blob/master/src/couch_index/src/couch_index_sup.erl#L24